samcli/lib/build/app_builder.py

```diff
@@ -676,7 +678,9 @@ def _get_build_options(
         """
 
         if metadata and dependency_manager and dependency_manager == "npm-esbuild":
-            build_props = metadata.get(BUILD_PROPERTIES, {})
+            # Make a copy of the metadata so that we don't force an updated build definition
+            # when we add the entry points
+            build_props = deepcopy(metadata.get(BUILD_PROPERTIES, {}))
```

Comment:
Looks like `ResourceMetadataNormalizer` is the method which changes values of `build_props`, can we move `deepcopy` into there for not affecting other users of that method?

Reply:
Updated to remove deepcopy. The ResourceMetadataNormalizer already returns a fresh copy so I just update that dict.